plugins/dead_code_elimination/dead_code_elimination_main.ml

```diff
@@ -125,9 +168,19 @@ Config.manpage [
   used. Internally, we translate a program into the SSA form, though
   it will not be seen outside";
 
-  `S "DEPENDENCIES";
-  `P "$(b,bap-plugin-api)(1)";
-
-  `S "SEE ALSO";
-  `P "$(b,bap-plugin-api)(1), $(b,bap-plugin-ssa)(1)";
-]
+    `S "DEPENDENCIES";
+    `P "$(b,bap-plugin-api)(1)";
+
+    `S "SEE ALSO";
+    `P "$(b,bap-plugin-api)(1), $(b,bap-plugin-ssa)(1)";
+  ];
+  let level =
+    let doc =
+      "An integer that a level of optimization, i.e. what variables
```

Comment:
Specifies the optimization level. The higher the value the more aggressive (and less safe) optimizations are applied. 

On level 1, we optimize only the synthetic code that was generated by the lifter. Since such code can't leave a scope of instruction it is not affected by the imprecision of a control flow graph. On level 2, we also move and optimize processor flags. This removes a significant amount of code and simplifies the program and is a fair compromise between safety and performance. (Since flags are rarely used non-locally). Finally, on level 3 we extend our analysis to all variables.